src/Microsoft.Data.Analysis/DataFrameColumns/StringDataFrameColumn.cs

```diff
@@ -79,6 +79,27 @@ public void Append(string value)
             Length++;
         }
 
+        /// <summary>
+        /// Applies a function to all values in the column, that are not null.
+        /// </summary>
```

Comment:
Why not null instead of using the isvalid?

Reply:
IsValid can be used here as well, however for StringDataFrame column it doesn't make a lot of sense (actualy IsValid works a little bit slower in this case). Using IsValid instead of checking for null dramaticaly improves performance in case of PrimitivesDataFrame columns without Nulls (most use cases), because such columns uses validity buffers and checking for Null is very expensive and can be skiped if NullCount is 0